pkg/server/server.go

```diff
@@ -797,8 +812,14 @@ func (s *ProxyServer) serveRecvBackend(backend Backend, stream agent.AgentServic
 				} else {
 					klog.V(5).InfoS("DIAL_CLS sent to frontend", "dialID", resp.Random)
 				}
-				s.PendingDial.Remove(resp.Random)
-				klog.V(5).InfoS("Close streaming", "agentID", agentID, "dialID", resp.Random)
+				if s.PendingDial.Remove(resp.Random) != nil {
+					klog.ErrorS(nil, "Dial closed prematurely",
```

Comment:
Same question here about whether logging error here could be misleading

Reply:
Changed message to `Dial terminated (DIAL_CLS) by backend`